### PR TITLE
:adhesive_bandage: Add google search retrieval as builtin tool

### PR DESCRIPTION
**Pull request by Grogdunn**
Added in Vertex client configuration the possibility to use GoogleSearchRetrieval as tool as grounding feature

https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/grounding

next up (on other PR) grounding support with "retrieval" tool, more studies needed.

### CODE REVIEW

**Comment by Grogdunn** [comment]
aligned with main

File: models/spring-ai-vertex-ai-gemini/src/main/java/org/springframework/ai/vertexai/gemini/VertexAiGeminiChatClient.java [hunk 1]
```diff
@@ -243,8 +235,19 @@ private GeminiRequest createGeminiRequest(Prompt prompt) {
 		}
 
 		// Add the enabled functions definitions to the request's tools parameter.
+
+		List<Tool> tools = new ArrayList<>();
 		if (!CollectionUtils.isEmpty(functionsForThisRequest)) {
-			List<Tool> tools = this.getFunctionTools(functionsForThisRequest);
+			tools.addAll(this.getFunctionTools(functionsForThisRequest));
+		}
+		if (((VertexAiGeminiChatOptions) prompt.getOptions()).getGoogleSearchRetrieval()) {
```
Comment (markpollack) [inline models/spring-ai-vertex-ai-gemini/src/main/java/org/springframework/ai/vertexai/gemini/VertexAiGeminiChatClient.java:243]: There is a NPE here when running the tests.   I have fixed it, but it means this code wasn't really verified.

In future, please add a test to verify the behavior.

File: models/spring-ai-vertex-ai-gemini/src/main/java/org/springframework/ai/vertexai/gemini/VertexAiGeminiChatOptions.java [hunk 2]
```diff
@@ -99,6 +96,12 @@ public enum TransportType {
 	@JsonIgnore
 	private Set<String> functions = new HashSet<>();
 
+	/**
+	 * Use Google search Grounding feature
+	 */
+	@JsonIgnore
+	private boolean googleSearchRetrieval = false;
```
Comment (markpollack) [inline models/spring-ai-vertex-ai-gemini/src/main/java/org/springframework/ai/vertexai/gemini/VertexAiGeminiChatOptions.java:103]: also needs to be part of the `fromOptions` method.

**Comment by markpollack** [comment]
merged in 793052c2e96cf95230ddd8f7bbcb535abdffa397

I had a bad merge conflict somehow and ended up redoing the changes and adding a test on another branch.  Sorry I forgot to add your name in the commit message.  next time! ;)

**Pull request closed without merging**